src/asammdf/blocks/mdf_v4.py

```diff
@@ -1586,25 +1655,29 @@ def _load_data(
                     new_data = memoryview(new_data)
 
                 if rm and invalidation_size:
+                    if invalidation_info is None:
+                        raise RuntimeError("'invalidation_info' cannot be None if 'rm and invalidation_size' is True")
+
                     if invalidation_info.all_valid:
                         count = original_size // samples_size
                         new_invalidation_data = b"\0" * (count * invalidation_size)
 
                     else:
                         seek(invalidation_info.address)
-                        new_invalidation_data = read(invalidation_info.size)
+                        original_size = typing.cast(int, invalidation_info.original_size)
```

Comment:
here we must use the `size` attribute since it is the size of the block that might be compressed. The `original_size` is the uncompressed size

Reply:
There is no `size` attribute for the class `InvalidationBlockInfo`, that is why I changed to `original_size`. This is another kind of error revealed by mypy, a non-existing attribute.

So I guess it should actually be `compressed_size`:

```suggestion
                        original_size = typing.cast(int, invalidation_info.compressed_size)
```

Reply:
Corect it should be `compressed_size`

Reply:
Oh wait, I know why. I am opening a PR right away :)